bika/lims/exportimport/dataimport.py

```diff
@@ -136,9 +143,10 @@ def getAnalysisServicesDisplayList(self):
             text to be displayed.
         '''
         bsc = getToolByName(self, 'bika_setup_catalog')
-        items = [('', '')] + [(o.getObject().Keyword, o.Title) for o in
-                                bsc(portal_type = 'AnalysisService',
-                                    is_active = True)]
+        brains = bsc(portal_type='AnalysisService')
+        items = []
+        for item in brains:
+            items.append((item.UID, item.Title))
```

Comment:
As far as I am concerned, this function is used by some instrument interfaces (e.g. Qubit, Symex XS, etc.) to know the analysis services to which the results have to be applied. With this change, the value of the display list becomes the UID of the service instead of the keyword, so the parsers will probably not work properly:

https://github.com/senaite/senaite.core/blob/26efddb95dc71fe61921cb93dc8d3603d72862fd/bika/lims/exportimport/instruments/sysmex/xs/i500.py#L71-L78

Setup catalog has `getKeyword` metadata, so I think the following would be fine:

```python
items = map(lambda item: (item.getKeyword, item.Title), brains)
```

Reply:
The change I made here was purely  flake8 cosmetics, the previous version also returned UID and Title.

Reply:
Hi @mikejmets , apologies for my slow reply. I've been out for two weeks.
Ummm... don't think so. Note the "ugly" previous implementation was using `getObject().Keyword`:

```python
items = [('', '')] + [(o.getObject().Keyword, o.Title) for o in
                                bsc(portal_type = 'AnalysisService',
                                    is_active = True)]
```

while your implementation uses `item.UID`:
```python
items.append((item.UID, item.Title))
```

Reply:
My bad, changed it to use getKeyword.